pkg/scheduler/framework/v1alpha1/registry.go

```diff
@@ -20,11 +20,36 @@ import (
 	"fmt"
 
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/json"
+	"k8s.io/apimachinery/pkg/util/yaml"
 )
 
 // PluginFactory is a function that builds a plugin.
 type PluginFactory = func(configuration *runtime.Unknown, f FrameworkHandle) (Plugin, error)
 
+// DecodeInto decodes configuration whose type is *runtime.Unknown to the interface into.
+func DecodeInto(configuration *runtime.Unknown, into interface{}) error {
+	if configuration == nil {
+		return nil
+	}
+
+	var data []byte
+	switch configuration.ContentType {
+	// If ContentType is empty, it means ContentTypeJSON by default.
+	case runtime.ContentTypeJSON, "":
+		data = configuration.Raw
+	case runtime.ContentTypeYAML:
+		var err error
+		if data, err = yaml.ToJSON(configuration.Raw); err != nil {
```

Comment:
why not decode directly?

Reply:
I think either one is OK. No strong tendency for it. Just follow the pattern  in https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/runtime/serializer/yaml/yaml.go#L40.

Reply:
TBH, I would change the code in apimachinery. It adds unnecessary processing.

Reply:
I tried to find the answer, and found it at http://ghodss.com/2014/the-right-way-to-handle-yaml-in-golang/. And https://github.com/kubernetes-sigs/yaml exists for this reason, `this library first converts YAML to JSON using go-yaml and then uses json.Marshal and json.Unmarshal to convert to or from the struct. `.

Reply:
It seems better to use `k8s.sigs.io/yaml` to unmashal yaml directly. Updated.

Reply:
Thanks. Much better to delegate this pattern to a single library.